primitives/session-validator-management/query/src/lib.rs

```diff
@@ -0,0 +1,211 @@
+pub mod commands;
+pub mod get_registrations;
+pub mod types;
+
+use async_trait::async_trait;
+use authority_selection_inherents::authority_selection_inputs::AuthoritySelectionInputs;
+use authority_selection_inherents::filter_invalid_candidates::CandidateValidationApi;
+use derive_new::new;
+use main_chain_follower_api::candidate::RawPermissionedCandidateData;
+use main_chain_follower_api::CandidateDataSource;
+use plutus::ToDatum;
+use sidechain_block_search::{predicates::AnyBlockInEpoch, FindSidechainBlock, SidechainInfo};
+use sidechain_domain::{MainchainPublicKey, McEpochNumber, ScEpochNumber};
+use sp_api::ProvideRuntimeApi;
+use sp_blockchain::{HeaderBackend, Info};
+use sp_core::bytes::to_hex;
+use sp_runtime::traits::NumberFor;
+use sp_runtime::traits::{Block as BlockT, Zero};
+use sp_session_validator_management::SessionValidatorManagementApi;
+use sp_sidechain::{GetSidechainParams, GetSidechainStatus};
+use std::sync::Arc;
+use types::*;
+
+pub type QueryResult<T> = Result<T, String>;
+
+#[async_trait]
```

Comment:
If you have time and want to play with it, since rust 1.75 async fn can be used in trait without the annotation. There are some differences though and I haven't played with it yet: https://blog.rust-lang.org/2023/12/21/async-fn-rpit-in-traits.html

Reply:
I am able to replace:
```
#[async_trait]
pub trait SessionValidatorManagementQueryApi {
```
with
```
#[trait_variant::make(SessionValidatorManagementQueryApi: Send)]
pub trait LocalSessionValidatorManagementQueryApi {
```
I'm not sure if it worth doing now. Pros: we'll use "more modern way" and will have some example for future generations in the code. Cons: we need to stick with `#[async_trait]` in the next layer, because traits defined with `#[rpc(...)]` require implementations to use `#[async_trait]` annotation anyway.

Reply:
I think that having such example in the code would be nice, up to you though.

Reply:
I'll do it in a separate PR in few minutes!